lib/puppet/face/node/clean.rb

```diff
@@ -1,40 +1,25 @@
 Puppet::Face.define(:node, '0.0.1') do
   action(:clean) do
 
-    summary _("Clean up signed certs, cached facts, node objects, and reports for a node stored by the puppetmaster")
+    summary _("Clean up cached facts, node objects, and reports for a node stored by the puppetmaster")
     arguments _("<host1> [<host2> ...]")
     description <<-'EOT'
       Cleans up the following information a puppet master knows about a node:
 
-      <Signed certificates> - ($vardir/ssl/ca/signed/node.domain.pem)
-
       <Cached facts> - ($vardir/yaml/facts/node.domain.yaml)
 
       <Cached node objects> - ($vardir/yaml/node/node.domain.yaml)
 
       <Reports> - ($vardir/reports/node.domain)
 
+      NOTE: this action no longer cleans up certs. For cert cleaning, please use `puppetserver ca clean`.
     EOT
 
     when_invoked do |*args|
       nodes = args[0..-2]
       options = args.last
       raise _("At least one node should be passed") if nodes.empty? || nodes == options
 
-      # This seems really bad; run_mode should be set as part of a class
-      # definition, and should not be modifiable beyond that.  This is one of
-      # the only places left in the code that tries to manipulate it. Other
-      # parts of code that handle certificates behave differently if the
-      # run_mode is master. Those other behaviors are needed for cleaning the
-      # certificates correctly.
-      Puppet.settings.preferred_run_mode = "master"
```

Comment:
This comment specifically calls out `run_mode = master` being necessary for cert behavior, but I'm curious if it isn't also needed to correctly determine vardir, reportdir, or yamldir?

Reply:
Yeah I was wondering about that too. Setting up something to try it.